tftp/backend.py

```diff
@@ -270,7 +336,16 @@ def get_reader(self, file_name):
             target_path = self.base.descendant(file_name.split(b"/"))
         except InsecurePath as e:
             raise AccessViolation("Insecure path: %s" % e)
-        return FilesystemReader(target_path)
+        try:
+            # Try using FilesystemReader first
+            return FilesystemReader(target_path)
+        except FileNotFound:
+            # Try using DynamicReader if dyn_handler was supplied
+            if self.dyn_handler is not None:
+                return DynamicReader(file_name, self.dyn_handler)
+            else:
+                raise
```

Comment:
Out of curiosity: do you really need to have a backend, that can both write to filesystem and "read" using a DynamicReader? Maybe you need to create a separate `IBackend` implementation, that always returns a `DynamicReader` in `get_reader` and leave `FilesystemSynchronousBackend` as it is now?